sretoolbox/utils/cache.py

```diff
@@ -0,0 +1,167 @@
+"""
+Caching helper decorator
+"""
+
+import functools
+import weakref
+
+_CACHES = weakref.WeakKeyDictionary()
+
+
+def static(key):
+    """
+    Decorates a class method to look up and cache the execution result.
+
+    Cache access is scoped by 2 things: class instance and `key`
+
+    :param key: static cache key
+    :type key: string
```

Comment:
docstrings <3

Reply:
I'll add the param docs to all other functions as well